src/common/database.c

```diff
@@ -1371,6 +1371,30 @@ static gboolean pid_is_alive(int pid)
   }
 #else
   pid_is_alive = !((kill(pid, 0) == -1) && errno == ESRCH);
+
+#ifdef __linux__
+  // If this is Linux, we can query /proc to see if the pid is
+  // actually a darktable instance.
+  if (pid_is_alive)
+  {
+    char filename[64];
+    char cmdline[MAX_PATH+1];
+    sprintf(filename, "/proc/%d/cmdline", pid);
+    FILE* f = fopen(filename, "r");
+    if (f) {
+      size_t size;
+      size = fread(cmdline, sizeof(char), MAX_PATH, f);
+      if (size > 0)
+      {
+        if (strstr(cmdline, "darktable") == NULL) {
```

Comment:
That will work as long the original dt executable contained that case-sensitive sub-string in the full path to the `argv[0]`.
I'm wondering if that is a safe assumption? Seems like it?